packages/core/saved-objects/core-saved-objects-server/model_versions.md

```diff
@@ -0,0 +1,529 @@
+stages of their lifecycle.
+
+The currently available schemas are:
+
+#### forwardCompatibility
+
+This is a new concept introduced by model versions. This schema is used for inter-version compatibility.
+
+When retrieving a savedObject document from an index, if the version of the document is higher than the latest version
+known of the Kibana instance, the document will go through the `forwardCompatibility` schema of the associated model version.
+
+These conversion mechanism shouldn't assert the data itself, and only strip unknown fields to convert the document to 
+the **shape** of the document at the given version.
+
+Basically, this schema should keep all the known fields of a given version, and remove all the unknown fields, without throwing.
+
+Usage examples are available in the `use-case examples` section below.
+
+**note:** *Even if highly recommended, implementing this schema is not strictly required. Type owners can manage unknown fields
+          and inter-version compatibility themselves in their service layer instead.*
+
+#### create
+
+This is a direct replacement for [the old SavedObjectType.schemas](https://github.com/elastic/kibana/blob/9b330e493216e8dde3166451e4714966f63f5ab7/packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts#L75-L82)
+definition, now directly included in the model version definition.
+
+As a refresher the `create` schema is a `@kbn/config-schema` object-type schema, and is used to validate the properties of the document
+during `create` and `bulkCreate` operations.
+
+**note:** *Implementing this schema is optional, but still recommended, as otherwise there will be no validating when 
+           importing objects*
+
+## Use-case examples
+
+These are example of the migration scenario currently supported (out of the box) by the system.
+
+**note:** *more complex scenarios (e.g field mutation by copy/sync) could already be implemented, but without
+          the proper tooling exposed from Core, most of the work related to sync and compatibility would have to be
+          implemented in the domain layer of the type owners, which is why we're not documenting them yet.*
+
+### Adding a non-indexed field without default value
+
+We are currently in model version 1, and our type has 2 indexed fields defined: `foo` and `bar`.
+
+The definition of the type at version 1 would look like:
+
+```ts
+const myType: SavedObjectsType = {
+  name: 'test',
+  namespaceType: 'single',
+  switchToModelVersionAt: '8.10.0',
+  modelVersions: {
+    // initial (and current) model version
+    1: {
+      changes: [],
+      schemas: {
+        // FC schema defining the known fields (indexed or not) for this version
+        forwardCompatibility: schema.object(
+          { foo: schema.string(), bar: schema.string() },
+          { unknowns: 'ignore' } // note the `unknown: ignore` which is how we're evicting the unknown fields
+        ),
+      },
```

Comment:
We should include `create` here too. Folks tend to "copy-paste" from examples and it'll remind them to fill it in. I know defining an on creation schema is optional, never the less, it's better to have it removed deliberately than forgetting to add one.